frame/nomination-pools/src/lib.rs

```diff
@@ -1514,9 +1556,6 @@ pub mod pallet {
 			let current_era = T::StakingInterface::current_era();
 			let unbond_era = T::StakingInterface::bonding_duration().saturating_add(current_era);
 
-			// Try and unbond in the member map.
-			member.try_unbond(unbonding_points, unbond_era)?;
```

Comment:
if this is transactional, why would it matter if this happens at the beginning or later?

Reply:
it is not about being last, it is about the first argument changing. If you unbond 10 points, you need to first liquidate that 10 to balance, then issue it back in the corresponding unbonding pool, and in return you get x points back, and that's the things that we need to track. 

In absence of slashing x is also 10, but not always. 

This was the main bug that is being fixed here.